src/diffusers/loaders/lora.py

```diff
@@ -1540,6 +1541,7 @@ def load_lora_into_transformer(cls, state_dict, transformer, adapter_name=None,
         state_dict = {
             k.replace(f"{cls.transformer_name}.", ""): v for k, v in state_dict.items() if k in transformer_keys
         }
+        state_dict = convert_unet_state_dict_to_peft(state_dict)
```

Comment:
Should this always be done or should this be conditional? Could we add some tests for this?

Reply:
I haven't really checked why is this happening, there's something in the training code (not sure where) that saves the loras in both formats, ideally we should fix that instead of doing something in the loading.

But yeah, I can check for the keys before doing it.

Reply:
Yeah better to add a check to be sure. 

These are the two places where we save things:

https://github.com/huggingface/diffusers/blob/23a2cd33379015b8d59f235dbf8879272175adf9/examples/dreambooth/train_dreambooth_lora_sd3.py#L1167C55-L1167C80

https://github.com/huggingface/diffusers/blob/23a2cd33379015b8d59f235dbf8879272175adf9/examples/dreambooth/train_dreambooth_lora_sd3.py#L1633

I don't see anything strange in here too:
https://github.com/huggingface/diffusers/blob/23a2cd33379015b8d59f235dbf8879272175adf9/src/diffusers/loaders/lora.py#L1645

So, not sure what you mean by 

> there's something in the training code (not sure where) that saves the loras in both formats

Reply:
oh I meant that the LoRAs that have this error, were trained with SimpleTuner, so I don't know if what provokes this is in the diffusers training code or if its something provoked by some modification done by SimpleTuner.

I couldn't find anything in our code either, also I trained a couple of test LoRAs and all of them were saved correctly. So I'm still not that convinced if we should do this.

Reply:
If that is the case, I believe we should add the check on the loading level.